test/journeys/lib/test-helpers/space-widget/meet.js

```diff
@@ -57,3 +63,101 @@ export function hangup(aBrowser) {
   aBrowser.waitForVisible(elements.callControls);
   aBrowser.element(elements.meetWidget).element(elements.hangupButton).click();
 }
+
+/**
+ * Test to hangup call before reciever answers
+ * @param {Object} browserLocal
+ * @param {Object} browserRemote
+ * @returns {void}
+ */
+export function hangupBeforeTest(browserLocal, browserRemote) {
+  switchToMeet(browserLocal);
+  call(browserLocal, browserRemote);
+  hangup(browserLocal);
+  browserRemote.element(elements.meetWidget).element(elements.callButton).waitForVisible();
+}
+
+/**
+ * Test to decline incoming call
+ * @param {Object} browserLocal
+ * @param {Object} browserRemote
+ * @returns {void}
+ */
+export function declineIncomingTest(browserLocal, browserRemote) {
+  switchToMeet(browserRemote);
+  call(browserRemote, browserLocal);
+  decline(browserLocal);
+  browserRemote.element(elements.meetWidget).element(elements.callButton).waitForVisible();
+  // Pausing to let locus session flush
+  browserLocal.pause(10000);
+}
+
+/**
+ * Test to hangup during ongoing call
+ * @param {Object} browserLocal
+ * @param {Object} browserRemote
+ * @returns {void}
+ */
+export function hangupDuringTest(browserLocal, browserRemote) {
+  clearEventLog(browserLocal);
+  clearEventLog(browserRemote);
+  switchToMeet(browserLocal);
+  call(browserLocal, browserRemote);
+  answer(browserRemote);
+  hangup(browserLocal);
+  browserLocal.waitForVisible(elements.messageWidget);
+  const result = browserLocal.execute(() => {
```

Comment:
Could you move these call event parsing code blocks into their own function. Looks like it is copy pasted in the `callEventTest` function.